modules/mauth-common/src/main/java/com/mdsol/mauth/util/MAuthSignatureHelper.java

```diff
@@ -224,35 +223,99 @@ public static String generateEncryptedQueryParams(String query) throws IOExcepti
         if (encryptedQueryParams.length() > 0) {
           encryptedQueryParams.append('&');
         }
-        encryptedQueryParams.append(urlEncodeValue(name)).append('=').append(urlEncodeValue(value));
+        encryptedQueryParams.append(encordQueryString(name)).append('=').append(encordQueryString(value));
       }
     }
 
     return encryptedQueryParams.toString();
   }
 
   /**
-   * encode a string value using `UTF-8` encoding scheme
+   * decode and sort a encoded query string
+   * @param encodedQuery a url-encoded string
+   * @return sorted decoded query parameters
+   */
+  private static String[] sortQueryParameters(String encodedQuery) {
+    String[] params = encodedQuery.split("&");
+    List<String> list = new ArrayList<String>();
+
+    for (String param : params) {
+      String [] keyPair = param.split("=");
+      if (keyPair.length > 0) {
+        String name = param.split("=")[0];
+        String value = keyPair.length > 1 ? param.split("=")[1] : "";
+        list.add(dencordQueryString(name) + "=" + dencordQueryString(value));
```

Comment:
Fixed param.split called 3 times.
Here is the requirement of query-string for mauth V2:
decoding the encoded query-string, sort by `decoded-key`=`decoded-value; 
rejoin the query-string as Url-encoded plus OAuth encodes;

See the latest change in https://github.com/mdsol/mauth-jvm-clients/commit/a98ad397ed2fed037490ef917234108f4f1bda97

Reply:
Yes I understand the logic, I was commenting on the code not handling an edge case when the key contains `=`.

Here's my implementation of the same logic that fixes it while being shorter and easier to understand (IMO):

```
    public static String generateEncryptedQueryParams(String encodedQuery) {
        if (encodedQuery == null || encodedQuery.isEmpty())
            return "";

      return Arrays.stream(encodedQuery.split("&"))
              .filter(s -> !s.isEmpty())
              .map(keyValStr -> {
                String[] split = keyValStr.split("=");
                String key = split[0];
                String value = split.length > 1 ? split[1] : "";
                return Pair.of(urlDecodeValue(key), urlDecodeValue(value));
              })
              .sorted()
              .map(keyVal -> urlEncodeValue(keyVal.getKey()) + "=" + urlEncodeValue(keyVal.getValue()))
              .collect(Collectors.joining("&"));
    }
```

Note that I've incorporated the logic of `decodeAndSortQueryParameters` too so that can be deleted.

Reply:
Thanks, updated as suggested in https://github.com/mdsol/mauth-jvm-clients/pull/122/commits/182a57384918e56538fc0cf3677140483e456023